aten/src/ATen/copy_wrapper.py

```diff
@@ -96,12 +96,12 @@
 }
 """)
 
-# Technically, no code should actually call _s_copy_from with a CPU self (this
-# only can happen when the src is CUDA from a CPU kernel) but for
-# completeness we fill out with a swap.
+# NB: Hypothetically, someone could call s_copy_from directly and get an error
+# message which claims something is not supported, when it actually is.  But
+# the correct fix in this case was to NOT call copy_from
 FUNCTION_FROM_SWAP = CodeTemplate("""\
 Tensor & ${Type}::_s_copy_from(const Tensor & src, Tensor & dst, bool non_blocking) const {
```

Comment:
I'm confused -- how is this function useful now?

Reply:
There's some hidden code that you can't see here which overrides `_s_copy_from` for CUDA tensors. So the general strategy is `_s_copy_from` is called from `s_copy` if a cpu-cuda conversion is involved; it's the double dispatch function.